File: server/src/utils/role.util.ts
/**
 * 与登录 JWT 中 roles 字段一致：来自 Role.roleCode（如 admin），
 * 兼容模板里使用的 R_SUPER / R_ADMIN 等标识。
 */
export function isSuperAdminRoles(roles: string[] | undefined | null): boolean {
  if (!roles?.length) return false;
  return (
    roles.includes("admin") ||
    roles.includes("R_SUPER") ||
    roles.includes("R_ADMIN")
  );
}

File: java-server/src/main/java/com/todothink/core/util/RoleUtil.java
package com.todothink.core.util;

import java.util.Arrays;
import java.util.List;

/**
 * 对齐 server/src/utils/role.util.ts
 */
public final class RoleUtil {

    private static final List<String> SUPER_ADMIN_CODES = Arrays.asList("admin", "R_SUPER", "R_ADMIN");

    private RoleUtil() {
    }

    public static boolean isSuperAdminRoles(List<String> roles) {
        if (roles == null || roles.isEmpty()) {
            return false;
        }
        for (String role : roles) {
            if (SUPER_ADMIN_CODES.contains(role)) {
                return true;
            }
        }
        return false;
    }
}
